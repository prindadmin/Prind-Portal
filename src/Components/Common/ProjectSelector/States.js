export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
