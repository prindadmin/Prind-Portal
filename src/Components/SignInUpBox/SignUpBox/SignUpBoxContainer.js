import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import * as auth from '../../../Reducers/auth'

import PageComponent from './SignUpBox'

const mapStatetoProps = state => {
  return {
    auth: state.auth,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    init: () => {
      dispatch(auth.init())
    },
    signUp: (userDetails, resolve, reject) => {
      dispatch(auth.signUp(userDetails, resolve, reject))
    }
  }
}

export default withRouter(connect(mapStatetoProps, mapDispatchToProps)(PageComponent))
