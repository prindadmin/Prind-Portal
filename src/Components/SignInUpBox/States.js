export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
