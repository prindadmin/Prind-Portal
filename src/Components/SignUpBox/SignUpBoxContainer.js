import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import * as authReducer from '../../Reducers/authReducer'

import PageComponent from './SignUpBox'

const mapStatetoProps = state => {
  return {
    auth: state.auth,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    init: () => {
      dispatch(authReducer.init())
    },
    signUp: (userDetails, resolve, reject) => {
      dispatch(authReducer.signUp(userDetails, resolve, reject))
    }
  }
}

export default withRouter(connect(mapStatetoProps, mapDispatchToProps)(PageComponent))
