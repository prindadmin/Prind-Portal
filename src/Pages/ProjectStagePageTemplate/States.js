export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
