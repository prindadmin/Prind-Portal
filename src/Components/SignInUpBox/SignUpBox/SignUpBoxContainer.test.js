import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import {
  HashRouter as Router
} from "react-router-dom";
import ReactGA from 'react-ga';

import * as Component from './SignUpBoxContainer'
import * as States from '../../../States'

import CanUseWebP from '../../../Functions/CheckIfWebpSupported'
jest.mock('../../../Functions/CheckIfWebpSupported')

// https://www.robinwieruch.de/react-connected-component-test

const mockStore = configureStore([]);

var component;

beforeEach(() => {
  CanUseWebP.mockReturnValue(true)
  ReactGA.initialize('dummy', { testMode: true });
  const store = mockStore({
    auth: {
      info: {},
      error: {},
      isSignedIn: States.AUTH_UNKNOWN,
      isConfirmed: States.AUTH_UNKNOWN,
      hasSignedUp: States.AUTH_UNKNOWN,
      hasSentCode: States.AUTH_UNKNOWN,
      hasChangedPassword: States.AUTH_UNKNOWN,
      passwordResetRequired: States.AUTH_UNKNOWN
    },
    user: {
      fetching: false,
      currentRoute: '/profile',
      details: {},
    }
  });

  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: function() { },
    history: {
      push: mockHistoryPush,
    },
  };

  component = renderer.create(
    <Provider store={store}>
      <Router>
        <Component.default {...props} />
      </Router>
    </Provider>
  );
});


it('should render with given state', () => {
  expect(component.toJSON()).toMatchSnapshot();
});
