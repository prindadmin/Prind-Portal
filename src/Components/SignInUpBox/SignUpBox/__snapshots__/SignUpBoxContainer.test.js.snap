// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with given state 1`] = `
Array [
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/prind-tech-logo.webp"
      />
    </a>
  </div>,
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>,
  <div
    className="welcome-text-body"
  >
    Fill in the form below to sign up to the Foundations portal
  </div>,
  <form
    className="sign-up-form form"
    onSubmit={[Function]}
  >
    <input
      className="default"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      className="default"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="default"
      id="firstName"
      name="firstName"
      onChange={[Function]}
      placeholder="First Name"
      type="text"
      value=""
    />
    <input
      className="default"
      id="lastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Surname"
      type="text"
      value=""
    />
    <input
      className="default"
      id="company"
      name="company"
      onChange={[Function]}
      placeholder="Company"
      type="text"
      value=""
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="signUpButton"
      onClick={[Function]}
      type="submit"
      value="Sign up"
    />
    <div
      className="spacer"
    />
    <p
      className="sign-up-in-text"
      onClick={[Function]}
    >
      Already have an account?  Click here to sign in
    </p>
  </form>,
]
`;
